demos/coreMQTT_Agent/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# FreeRTOS coreMQTT demos
+afr_demo_module(core_mqtt_agent)
+
+# Set the default demo that is enabled in the downloaded package.
+afr_set_demo_metadata(ID "CORE_MQTT_CONNECTION_SHARING_DEMO")
+afr_set_demo_metadata(DESCRIPTION "Examples that demonstrate the MQTT library")
+afr_set_demo_metadata(DISPLAY_NAME "coreMQTT Demos")
+
+# Add the CMakeLists.txt file of module to metadata list.
+afr_module_cmake_files(${AFR_CURRENT_MODULE} 
+    ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt
+)
+
+afr_module_sources(
+    ${AFR_CURRENT_MODULE}
+    INTERFACE
+        "${CMAKE_CURRENT_LIST_DIR}/mqtt_demo_connection_sharing.c"
+        # Add the header file added to the target's metadata so that it
+        # is available in code downloaded from the FreeRTOS console.
+        ${AFR_DEMOS_DIR}/include/aws_iot_metrics.h
+)
+
+afr_module_include_dirs(
+    ${AFR_CURRENT_MODULE}
+    INTERFACE
+       "${AFR_DEMOS_DIR}/include"
+)
+
+afr_module_dependencies(
+    ${AFR_CURRENT_MODULE}
+    INTERFACE
```

Comment:
There needs to be a dependency on `AFR::mqtt_agent_interface` module for the interface implementation files

Reply:
Similar dependency needs to be added to the `core_mqtt_agent_demo_dependencies` target as well

Reply:
Added